apps/web/src/app/[orgShortcode]/settings/user/security/page.tsx

```diff
@@ -22,8 +22,61 @@ import { PageTitle } from '../../_components/page-title';
 import { Skeleton } from '@/src/components/shadcn-ui/skeleton';
 import { Switch } from '@/src/components/shadcn-ui/switch';
 import { Button } from '@/src/components/shadcn-ui/button';
+import { Input } from '@/src/components/shadcn-ui/input';
 // import { PasskeyNameModal } from './_components/passkey-modals';
 
+import { CheckCircle } from '@phosphor-icons/react';
+
+export function RecoveryEmailSection() {
+  const [recoveryEmail, setRecoveryEmail] = useState('');
+
+  const { data: recoveryEmailStatus } =
+    platform.account.security.getRecoveryEmailStatus.useQuery();
+  const { mutateAsync: setRecoveryEmailMutation } =
+    platform.account.security.setRecoveryEmail.useMutation();
+
+  const handleSetRecoveryEmail = async () => {
+    toast.promise(setRecoveryEmailMutation({ recoveryEmail }), {
+      loading: 'Sending you a verification code to your email',
+      success: 'Check your emails for a verification code',
+      error: 'Something went wrong while setting your recovery email'
+    });
+  };
+
+  return (
+    <div className="space-y-2">
+      <h3 className="text-lg font-medium">Recovery Email</h3>
+      {recoveryEmailStatus?.isSet ? (
+        recoveryEmailStatus.isVerified && (
+          <div className="flex items-center space-x-1 text-green-600">
+            <CheckCircle
+              size={20}
+              weight="fill"
+            />
+            <p>Your recovery email is set and verified.</p>
+          </div>
+        )
+      ) : (
+        <div>
+          <Input
+            label="Recovery Email"
+            type="email"
+            value={recoveryEmail}
+            onChange={(e) => setRecoveryEmail(e.target.value)}
+            placeholder="Enter recovery email"
+            onKeyDown={(e) => {
+              if (e.key === 'Enter') {
+                void handleSetRecoveryEmail();
+              }
+            }}
+          />
+          <Button onClick={handleSetRecoveryEmail}>Set Recovery Email</Button>
+        </div>
+      )}
```

Comment:
There is no state here that represents the condition when the mail is sent but not verified yet, that would prevent users from knowing that they need to verify the email.
also there is no way for user to edit the email before verifying incase they have set a wrong email while sending the link